src/sync/sys/unix/net.rs

```diff
@@ -0,0 +1,333 @@
+/*
+	Copyright The containerd Authors.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+use crate::error::Result;
+use nix::sys::socket::*;
+use std::io::{self};
+use std::os::unix::io::RawFd;
+use std::os::unix::prelude::AsRawFd;
+use nix::Error;
+
+use nix::unistd::*;
+use std::sync::{Arc};
+use std::sync::atomic::{AtomicBool, Ordering};
+use crate::common::{self, client_connect, SOCK_CLOEXEC};
+#[cfg(target_os = "macos")] 
+use crate::common::set_fd_close_exec;
+use nix::sys::socket::{self};
+
+pub struct PipeListener {
+    fd: RawFd,
+    monitor_fd: (RawFd, RawFd),
+}
+
+impl AsRawFd for PipeListener {
+    fn as_raw_fd(&self) -> RawFd {
+        self.fd
+    }
+}
+
+impl PipeListener {
+    pub(crate) fn new(sockaddr: &str) -> Result<PipeListener> {
+        let (fd, _) = common::do_bind(sockaddr)?;
+        common::do_listen(fd)?;
+
+        let fds = PipeListener::new_monitor_fd()?;
+
+        Ok(PipeListener {
+            fd,
+            monitor_fd: fds,
+        })
+    }
+
+    pub(crate) fn new_from_fd(fd: RawFd) -> Result<PipeListener> {
+        let fds = PipeListener::new_monitor_fd()?;
+
+        Ok(PipeListener {
+            fd,
+            monitor_fd: fds,
+        })
+    }
+
+    fn new_monitor_fd() ->  Result<(i32, i32)> {
+        #[cfg(target_os = "linux")]
+        let fds = pipe2(nix::fcntl::OFlag::O_CLOEXEC)?;
+ 
+        
+        #[cfg(target_os = "macos")] 
+        let fds = {
+            let (rfd, wfd) = pipe()?;
+            set_fd_close_exec(rfd)?;
+            set_fd_close_exec(wfd)?;
+            (rfd, wfd)
+        };
```

Comment:
I think it's better to wrap the `Macos` or `Linux` sys call into separate files such as `***_macos.rs` and `***_linux.rs` for `pipe()` and `accept()` blew.

Reply:
have a separate implementation of the all the new types or still have the `unix` folder with base implementation but just have additional files for these couple calls?

Reply:
I think it's better to have a separate folder.

Reply:
I started to look into this but I am not sure if this PR is the right place to do it. This is pretty much a straight copy of the existing logic. I might be missing something but breaking it out to its individual folders would cause a lot of duplication logic that is shared between mac/linux.

There is also quite a bit of this type of conditional compilation that is similar to this in https://github.com/containerd/ttrpc-rust/blob/master/src/common.rs.  That should probably be cleaned up too.

I would propose that type clean up would be better done outside of the refactoring. thoughts?

Reply:
Ok, I agree with you.